### ISSUE TITLE
pinDigest is also used in other managers than docker

### ISSUE THREAD

**Issue by dirkoberhaus** [renovatebot/renovate#14177]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.79.1

### If you're self-hosting Renovate, tell us what version of the platform you run.

Linux (Docker container)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

We have set "pinDigests": true in our renovate json to pin used docker images in Dockerfile etc.
Since 2-3 Weeks our renovate job logs following errors:

Error updating branch: update failure .... 

The expected updates are in .gitlab-ci.yml includes.

We invested some time and it seems that in the function confirmIfDepUpdated
is looking for digests where no digests are supported. For example in .gitlab-ci.yml includes.

Is there a config change we missed or is this a bug?

### Relevant debug logs

<details><summary>Logs</summary>

```
{"name":"renovate","hostname":"xxxx","pid":39,"level":40,"logContext":"Go76X9aINXdc1TGggar3U","repository":"xxxx","branch":"renovate/pin-dependencies","msg":"Error updating branch: update failure","time":"2022-02-11T16:18:46.538Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by dirkoberhaus** [comment]
Hi, I've created a minimal repo to reproduce the error:

https://gitlab.com/oberhaus77/renovate-test

Kind regards

**Comment by rarkins** [comment]
Thanks for the reproduction. Forked to https://github.com/renovate-reproductions/14177

The `pinDigests` *is* intended to be usable by other managers though, so the title of this bug report isn't the right one. Please expand the title/description so that it's clearer on the problem, e.g. does "Error updating gitlabci includes if pinDigests=true" describe your problem?

**Comment by dirkoberhaus** [comment]
Yes, this is the problem I've seen with Renovate version 31.81.7

The last version without that problem in our Renovate job seems to be 28.26.0.

**Comment by rarkins** [comment]
As a workaround, configure `pinDigests` for only Docker this way:

```
{
  "docker": {
    "pinDigests": true
  }
}
```

**Comment by therufuser** [comment]
Hey, I unfortunately only found this github-issue after spending some time debugging it. This matches what I've been able to find out.

My Workaround was the opposite:
```
{
    "gitlabci-include": {
        "pinDigests": false
    }
}
```

After bisecting the renovate-history I found the version which introduced digest-extraction for GitLab-repositories and this behaviour. It's v31.17.0 created by this PR: https://github.com/renovatebot/renovate/pull/13361

**Comment by therufuser** [comment]
@hasanwhitesource Could something like #15113 work for `gitlabci-include` as well?
If so I might create a PR next time I'm at work.

**Comment by viceice** [comment]
That manager will be merged to gitlabci soon.

**Comment by ppussar** [comment]
@therufuser @rarkins just to be sure, the workaround was to switch off docker-digests pinning in gitlab-ci? 
Or is there any chance to get it working?

**Comment by craig** [comment]
>That manager will be merged to gitlabci soon. so it will fixed automatically
I think it's still not fixed?

**Comment by prehor** [comment]
I have similar problem with HelmRepository `type:oci` which [uses docker datasource](https://docs.renovatebot.com/modules/manager/flux/#helmrelease-support) and `"extends": ["docker:pinDigest"]`. [@rarkins workaround](#issuecomment-1040231401) helps in this situation too:

> As a workaround, configure `pinDigests` for only Docker this way:
> 
> ```
> {
>   "docker": {
>     "pinDigests": true
>   }
> }
> ```

**Comment by daylicron** [comment]
We have the same issue regarding gitlabci-include and had to use the workaround
```
  "gitlabci-include": {
        "pinDigests": false,
    },
```
Is there anything new here? A fix would be highly appreciated.
